plugin-taskbar/lxqttaskbutton.cpp

```diff
@@ -582,6 +583,49 @@ void LXQtTaskButton::moveApplicationToPrevNextDesktop(bool next)
     KWindowSystem::setOnDesktop(mWindow, targetDesk);
 }
 
+/************************************************
+
+ ************************************************/
+void LXQtTaskButton::moveApplicationToPrevNextMonitor(bool next)
+{
+    KWindowInfo info(mWindow, NET::WMDesktop);
+    if (!info.isOnCurrentDesktop())
+        KWindowSystem::setCurrentDesktop(info.desktop());
+    if (isMinimized())
+        KWindowSystem::unminimizeWindow(mWindow);
+    KWindowSystem::forceActiveWindow(mWindow);
+    const QRect& windowGeometry = KWindowInfo(mWindow, NET::WMFrameExtents).frameGeometry();
+    QList<QScreen *> screens = QGuiApplication::screens();
+    if (screens.size() > 1){
+        for (int i = 0; i < screens.size(); ++i)
+        {
+            QRect screenGeometry = screens[i]->geometry();
+            if (screenGeometry.intersects(windowGeometry))
+            {
+                int targetScreen = i + (next ? 1 : -1);
+                if (targetScreen < 0)
+                    targetScreen += screens.size();
+                else if (targetScreen >= screens.size())
+                    targetScreen -= screens.size();
+                QRect targetScreenGeometry = screens[targetScreen]->geometry();
+                int X = windowGeometry.x() - screenGeometry.x() + targetScreenGeometry.x();
+                int Y = windowGeometry.y() - screenGeometry.y() + targetScreenGeometry.y();
+                NET::States state = KWindowInfo(mWindow, NET::WMState).state();
+                //      NW geometry |     x/y      |  from panel
+                const int flags = 1 | (0b011 << 8) | (0b011 << 12);
+                KWindowSystem::clearState(mWindow, NET::MaxHoriz | NET::MaxVert | NET::Max | NET::FullScreen);
+                NETRootInfo(QX11Info::connection(), 0, NET::WM2MoveResizeWindow).moveResizeWindowRequest(mWindow, flags, X, Y, 0, 0);            
+                QTimer::singleShot(200, [=]()
```

Comment:
This caught my eye before a full review. Please use a context here:
```c++
QTimer::singleShot(200, this, [=]()
```
Without it, a crash is possible, theoretically speaking.